ezyhttp-core/src/main/java/com/tvd12/ezyhttp/core/resources/ActualContentTypeDetector.java

```diff
@@ -0,0 +1,46 @@
+package com.tvd12.ezyhttp.core.resources;
+
+import com.tvd12.ezyhttp.core.constant.ContentTypes;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class ActualContentTypeDetector {
```

Comment:
1. Các trường trong lớp singleton nên được khai báo là biến thông thường thay vì static cho đơn giản và ngắn gọn, vì chúng chỉ được khởi tạo một lần khi instance được tạo.
2. Sẽ có trường hợp file không có extension, ví dụ `/hello.word.you/file`

```java
package com.tvd12.ezyhttp.core.resources;

import com.tvd12.ezyfox.collect.Sets;
import com.tvd12.ezyfox.util.EzyMapBuilder;
import com.tvd12.ezyhttp.core.constant.ContentTypes;

import java.util.Map;
import java.util.Set;

public class ActualContentTypeDetector {
    
    private final Set<String> needToDetectActualContentTypes;
    private final Map<String, String> actualContentTypeByExtension;
    private static final ActualContentTypeDetector INSTANCE =
        new ActualContentTypeDetector();

    private ActualContentTypeDetector() {
        this.needToDetectActualContentTypes = Sets.newHashSet(
            ContentTypes.GZIP
        );
        this.actualContentTypeByExtension = EzyMapBuilder.mapBuilder()
            .put("wasm.gz", ContentTypes.APPLICATION_WASM)
            .toMap();
    }
    
    public static ActualContentTypeDetector getInstance() {
        return INSTANCE;
    }

    public String detect(
        String resourcePath,
        String originalContentType
    ) {
        if (!needToDetectActualContentTypes.contains(originalContentType)) {
            return originalContentType;
        }
        String extension = extractExtension(resourcePath);
        return actualContentTypeByExtension.getOrDefault(
            extension,
            originalContentType
        );
    }

    private String extractExtension(String resourcePath) {
        int dotCount = 0;
        int startExtensionIndex = 0;
        for (int i = resourcePath.length() - 1; i >= 0; --i) {
            char ch = resourcePath.charAt(i);
            if (ch == '.') {
                ++dotCount;
            }
            
            // if got second dot count or start of file name
            if (dotCount == 2 || ch == '/' || ch == '\\') {
                startExtensionIndex = i + 1;
                break;
            }
        }
        return resourcePath.substring(startExtensionIndex);
    }
}
```

Reply:
Trường hợp `/hello.word.you/file` hàm extractExtension của anh sẽ trả về `file`. Em nghĩ đây là trường hợp không có extension nên trả về empty string